backend/inspirehep/records/cli.py

```diff
@@ -350,37 +350,65 @@ def legacy_records():
 
 
 @legacy_records.command(help="Upload marcxml column to S3.")
+@click.option(
+    "--task-queue",
+    help="Celery task queue",
+    default="xml_export",
+    show_default=True,
+)
 @click.option(
     "--bucket",
     help="S3 bucket where the files should be stored",
     default="inspire-qa-legacy",
     show_default=True,
 )
+@click.option(
+    "--bucket-size",
+    help="Size of database fetches",
+    default=100000,
+    show_default=True,
+)
 @click.option(
     "--batch-size",
+    help="Rows to export per celery task",
+    default=1000,
+    show_default=True,
+)
+@click.option(
+    "--db-batch-size",
     help="Size of database fetches",
     default=1000,
     show_default=True,
 )
 @with_appcontext
-def export_and_upload_xmls(bucket, batch_size):
-    upload_counter = 0
-    mime_type = "application/xml"
-    acl = current_app.config["S3_FILE_ACL"]
-
-    query = LegacyRecordsMirror.query.yield_per(batch_size)
-    for row in query:
-        filename = str(row.recid)
-        key = f"legacy_xml_{row.recid}"
-        fileob = BytesIO(row.marcxml)
-        try:
-            current_s3_instance.upload_file(
-                fileob, key, filename, mime_type, acl, bucket
-            )
-            upload_counter += 1
-        except ClientError:
-            LOGGER.warning("There was an issue uploading to s3", filename=filename)
-        if upload_counter % batch_size == 0:
-            LOGGER.info("files uploaded", upload_count=upload_counter)
-            LOGGER.info("latest upload", filename=filename)
-    LOGGER.info("Uploaded completed.", upload_count=upload_counter)
+def export_and_upload_xmls(task_queue, bucket, bucket_size, batch_size, db_batch_size):
+    row_count = LegacyRecordsMirror.query.count()
+    if batch_size > bucket_size:
+        batch_size = bucket_size
+    bucket_count = math.ceil(row_count / bucket_size)
+    export_tasks = []
+    LOGGER.info(
+        "Start export_and_upload_xmls",
+        bucket_prefix=bucket,
+        bucket_size=bucket_size,
+        batch_size=batch_size,
+        db_batch_size=db_batch_size,
+        bucket_count=bucket_count,
+    )
+    recid_chunks = chunker(
+        LegacyRecordsMirror.query.with_entities(LegacyRecordsMirror.recid).yield_per(
+            db_batch_size
+        ),
+        batch_size,
+    )
+    for count, recid_chunk in enumerate(recid_chunks):
+        bucket_name = f"{bucket}-{(count % bucket_count)}"
+        export_task = export_legacy_recids.s(
+            bucket_name, flatten_list(recid_chunk), db_batch_size
+        ).set(queue=task_queue)
+        export_tasks.append(export_task)
+
+    export_task_group = group(export_tasks)
+    export_group_result = export_task_group()
+    export_group_result.join()
```

Comment:
Why you're grouping tasks? Is there any particular reason for that? In my opinion, you can create tasks with `export_legacy_recids.delay()`, grouping in this case is not needed (without it you'll create ~12000 tasks, which is totally fine) and makes the code more complex. You can just send a task with a list of recids and then no need for chunking in the task.

Reply:
I used grouping to know when every task is finished, so the script only finishes if every task comes to an end. I agree this might cause problems so yeah i will remove it and use `.delay()`. 
But i dont really understand why the chunking is not needed, i thought its quiet useful to get a list of for example 1000 recids each iteration.